numpy/core/src/umath/string_ufuncs.cpp

```diff
@@ -355,6 +401,76 @@ init_comparison(PyObject *umath)
 }
 
 
+static NPY_CASTING
+string_addition_resolve_descriptors(
+        PyArrayMethodObject *NPY_UNUSED(self),
+        PyArray_DTypeMeta *NPY_UNUSED(dtypes[3]),
+        PyArray_Descr *given_descrs[3],
+        PyArray_Descr *loop_descrs[3],
+        npy_intp *NPY_UNUSED(view_offset))
+{
+    Py_INCREF(given_descrs[0]);
+    loop_descrs[0] = NPY_DT_CALL_ensure_canonical(given_descrs[0]);
+    Py_INCREF(given_descrs[1]);
+    loop_descrs[1] = NPY_DT_CALL_ensure_canonical(given_descrs[1]);
+    loop_descrs[2] = PyArray_DescrNew(given_descrs[0]);
```

Comment:
The test fails because this is clearly nonsense, the missing part of the test is mixing the byte order.

Reply:
I don't understand why this is nonsense. Could you please explain why and what's the correct way to do it?

Reply:
You are copying `given_descrs[0]` not `loop_descrs[0]` which is swapped.

Reply:
Right, I figured that right before your comments and was calling `NPY_DT_CALL_ensure_canonical` on `loop_descrs[2]`. But copying `loop_descrs[0]` directly makes more sense.